Sources/API/Converter.swift

```diff
@@ -48,4 +48,990 @@ enum Converter {
+//            );
+//        });
+//
+//        const textNode = RGATreeSplitNode.create(
+//            fromTextNodeID(pbTextNode.getId()!),
+//            richTextValue,
+//        );
+//        textNode.remove(fromTimeTicket(pbTextNode.getRemovedAt()));
+//        return textNode;
+//    }
+    
+    /**
+     * `fromRichText` converts the given Protobuf format to model format.
+     */
+//    function fromRichText<A>(pbText: PbJSONElement.RichText): CRDTRichText<A> {
+//        const rgaTreeSplit = new RGATreeSplit<RichTextValue>();
+//
+//        let prev = rgaTreeSplit.getHead();
+//        for (const pbNode of pbText.getNodesList()) {
+//            const current = rgaTreeSplit.insertAfter(prev, fromRichTextNode(pbNode));
+//            if (pbNode.hasInsPrevId()) {
+//                current.setInsPrev(
+//                    rgaTreeSplit.findNode(fromTextNodeID(pbNode.getInsPrevId()!)),
+//                );
+//            }
+//            prev = current;
+//        }
+//        const text = new CRDTRichText<A>(
+//            rgaTreeSplit,
+//            fromTimeTicket(pbText.getCreatedAt())!,
+//        );
+//        text.setMovedAt(fromTimeTicket(pbText.getMovedAt()));
+//        text.setRemovedAt(fromTimeTicket(pbText.getRemovedAt()));
+//        return text;
+//    }
+}
+
+// MARK: ChangePack
+extension Converter {
+    /**
+     * `toChangePack` converts the given model to Protobuf format.
+     */
+    static func toChangePack(pack: ChangePack) -> PbChangePack {
+        var pbChangePack = PbChangePack();
+        pbChangePack.documentKey = pack.getDocumentKey()
+        pbChangePack.checkpoint = toCheckpoint(pack.getCheckpoint())
+        pbChangePack.changes = toChanges(pack.getChanges())
+        // TODO: snapshot can be nil!
+        pbChangePack.snapshot = pack.getSnapshot() ?? Data()
+        if let minSyncedTicket = pack.getMinSyncedTicket() {
+            pbChangePack.minSyncedTicket = toTimeTicket(minSyncedTicket)
+        } else {
+            pbChangePack.clearMinSyncedTicket()
+        }
+        return pbChangePack;
+    }
+    
+    /**
+     * `fromChangePack` converts the given Protobuf format to model format.
+     */
+    static func fromChangePack(_ pbPack: PbChangePack) throws -> ChangePack {
+        ChangePack(key: pbPack.documentKey,
+                   checkpoint: fromCheckpoint(pbPack.checkpoint),
+                   changes: try fromChanges(pbPack.changes),
+                   snapshot: pbPack.snapshot,
+                   minSyncedTicket: pbPack.hasMinSyncedTicket ? fromTimeTicket(pbPack.minSyncedTicket) : nil)
+    }
+
+}
+
+// MARK: Change
+extension Converter {
+    /**
+     * `toChange` converts the given model to Protobuf format.
+     */
+    static func toChange(_ change: Change) -> PbChange {
+        var pbChange = PbChange();
+        pbChange.id = toChangeID(change.getID())
+        // TODO: message can be nil!
+        pbChange.message = change.getMessage() ?? ""
```

Comment:
'message' of Change is optional, but PbChange's is required. 
We need the default value of snapshot or change PbChange's to optional. 

@hackerwins

Reply:
Handle empty string as nil.